File: app/src/androidTest/java/de/werkelmann/rekote/ApplicationTest.java
package de.werkelmann.rekote;

import android.app.Application;
import android.test.ApplicationTestCase;
import android.test.suitebuilder.annotation.SmallTest;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.action.ViewActions.click;
import static android.support.test.espresso.assertion.ViewAssertions.matches;
import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
import static android.support.test.espresso.matcher.ViewMatchers.withId;

/**
 * <a href="http://d.android.com/tools/testing/testing_android.html">Testing Fundamentals</a>
 */
public class ApplicationTest extends ApplicationTestCase<Application> {
    public ApplicationTest() {
        super(Application.class);
    }

    @SmallTest
    public void makeTestSuiteNotEmpty() {
        onView(withId(R.id.action_settings))
                .perform(click())
                .check(matches(isDisplayed()));
    }
}

File: app/src/androidTest/kotlin/de/werkelmann/rekote/ApplicationTest.kt
package de.werkelmann.rekote

import android.app.Application
import android.support.test.espresso.Espresso.onView
import android.support.test.espresso.action.ViewActions.click
import android.support.test.espresso.assertion.ViewAssertions.matches
import android.support.test.espresso.matcher.ViewMatchers.isDisplayed
import android.support.test.espresso.matcher.ViewMatchers.withId
import android.test.ApplicationTestCase
import android.test.suitebuilder.annotation.SmallTest

/**
 * [Testing Fundamentals](http://d.android.com/tools/testing/testing_android.html)
 */
class ApplicationTest : ApplicationTestCase<Application>(Application::class.java) {

    @SmallTest
    fun makeTestSuiteNotEmpty() {
        onView(withId(R.id.action_settings))
                .perform(click())
                .check(matches(isDisplayed()))
    }
}